index.js

```diff
@@ -2,87 +2,57 @@ const {getOctokit, context} = require('@actions/github');
-			pushedTag = process.env.GITHUB_REF.replace('refs/tags/', '');
-			core.info('Using pushed tag as reference: ' + pushedTag);
-		}
-
-		// Get range to generate diff
-		let range = tags[1] + '..' + pushedTag;
-		if (tags.length < 2) {
-			const {stdout: rootCommit} = await execFile('git', ['rev-list', '--max-parents=0', 'HEAD']);
-			range = rootCommit.trim('') + '..' + pushedTag;
-		}
+	if (tags.length === 0) {
+		core.info('There is nothing to be done here. Exiting!');
+		return;
+	}
 
-		core.info('Computed range: ' + range);
+	let pushedTag = core.getInput('tag') || tags[0];
 
-		// Get commits between computed range
-		let {stdout: commits} = await execFile('git', ['log', '--format=%H%s', range]);
-		commits = commits.split('\n').filter(Boolean).map(line => ({
-			hash: line.slice(0, 8),
-			title: line.slice(40)
-		}));
+	if (process.env.GITHUB_REF.startsWith('refs/tags/')) {
+		pushedTag = process.env.GITHUB_REF.replace('refs/tags/', '');
+		core.info('Using pushed tag as reference: ' + pushedTag);
+	}
 
-		if (exclude) {
-			const regex = new RegExp(exclude);
-			commits = commits.filter(({title}) => !regex.test(title));
-		}
+	// Get range to generate diff
+	let range = tags[1] + '..' + pushedTag;
+	if (tags.length < 2) {
+		const {stdout: rootCommit} = await execFile('git', ['rev-list', '--max-parents=0', 'HEAD']);
+		range = rootCommit.trim('') + '..' + pushedTag;
+	}
 
-		// Generate markdown content
-		const commitEntries = [];
-		if (commits.length === 0) {
-			commitEntries.push('_Maintenance release_');
-		} else {
-			for (const {hash, title} of commits) {
-				const line = commitTemplate
-					.replace('{hash}', hash)
-					.replace('{title}', title)
-					.replace('{url}', repoURL + '/commit/' + hash);
-				commitEntries.push(line);
-			}
-		}
+	core.info('Computed range: ' + range);
 
-		const octokit = getOctokit(core.getInput('token'));
-		const createReleaseResponse = await octokit.repos.createRelease({
-			repo,
-			owner,
-			tag_name: pushedTag, // eslint-disable-line camelcase
-			body: releaseTemplate
-				.replace('{commits}', commitEntries.join('\n'))
-				.replace('{range}', `[\`${range}\`](${repoURL}/compare/${range})`),
-			draft: false,
-			prerelease: false
-		});
+	// Create a release with markdown content in body
+	const octokit = getOctokit(core.getInput('token'));
+	const createReleaseResponse = await octokit.repos.createRelease({
+		repo,
+		owner,
+		tag_name: pushedTag, // eslint-disable-line camelcase
+		body: await generateReleaseNotes({range, exclude, commitTemplate, releaseTemplate}),
+		draft: false,
+		prerelease: false
+	});
 
-		core.info('Created release `' + createReleaseResponse.data.id + '` for tag `' + pushedTag + '`');
-	} catch (error) {
-		core.setFailed(error.message);
-	}
+	core.info('Created release `' + createReleaseResponse.data.id + '` for tag `' + pushedTag + '`');
 }
 
-run();
+run().catch(error => {
+	core.setFailed(error.message);
+});
```

Comment:
The PR includes some unrelated changes though, it would be easier to leave them aside (like this try/catch and the package.json cleanup)